Meadowlark-js/backends/meadowlark-aws-dynamodb-opensearch/src/BaseDynamoRepository.ts

```diff
@@ -63,74 +81,66 @@ export function assignableSortKeyFromId(id: string): string {
   return `ASSIGN#ID#${id}`;
 }
 
-export function sortKeyFromEntityIdentity(entityInfo: EntityInfo): string {
-  return sortKeyFromId(documentIdForEntityInfo(entityInfo));
+export function sortKeyFromEntityIdentity(documentInfo: DocumentInfo): string {
+  return sortKeyFromId(documentIdForEntityInfo(documentInfo));
 }
 
-export function conditionCheckFrom(entityIdentifyingInfo: EntityIdentifyingInfo) {
-  invariant(
-    entityIdentifyingInfo.naturalKey.startsWith('NK#'),
-    `entityIdentifyingInfo.naturalKey "${entityIdentifyingInfo.naturalKey}" did not start with "NK#"`,
-  );
+export function conditionCheckFrom(entityIdentifyingInfo: DocumentIdentifyingInfo) {
   return {
     ConditionCheck: {
       TableName: tableOpts.tableName,
       Key: {
         pk: entityTypeStringFrom(entityIdentifyingInfo),
-        sk: sortKeyFromId(documentIdFromNaturalKeyString(entityIdentifyingInfo.naturalKey)),
+        sk: sortKeyFromId(idFromDocumentElements(entityIdentifyingInfo.documentIdentity)),
       },
       ConditionExpression: 'attribute_exists(sk)',
     },
   };
 }
 
-export function conditionCheckFromAssignable(assignableToTypeInfo: EntityTypeInfo, foreignKey: ReferentialConstraint) {
-  invariant(
-    foreignKey.constraintKey.startsWith('NK#'),
-    `foreignKey.constraintKey "${foreignKey.constraintKey}" did not start with "NK#"`,
-  );
+export function conditionCheckFromAssignable(assignableToTypeInfo: DocumentTypeInfo, documentIdentity: DocumentIdentity) {
   return {
     ConditionCheck: {
       TableName: tableOpts.tableName,
       Key: {
         pk: entityTypeStringFrom(assignableToTypeInfo),
-        sk: assignableSortKeyFromId(documentIdFromNaturalKeyString(foreignKey.constraintKey)),
+        sk: assignableSortKeyFromId(idFromDocumentElements(documentIdentity)),
       },
       ConditionExpression: 'attribute_exists(sk)',
     },
   };
 }
 
-export function foreignKeyConditions(entityInfo: EntityInfo): TransactWriteItem[] {
-  return entityInfo.foreignKeys.map((foreignKey) => {
-    const entityTypeInfo: EntityTypeInfo = {
+export function foreignKeyConditions(documentInfo: DocumentInfo): TransactWriteItem[] {
+  return documentInfo.foreignKeys.map((documentReference) => {
+    const entityTypeInfo: DocumentTypeInfo = {
       // TODO: Note for the future, this assumes the referenced entity is in the same project/namespace as the referring one
-      projectName: entityInfo.projectName,
-      projectVersion: entityInfo.projectVersion,
-      entityName: foreignKey.metaEdName,
+      projectName: documentInfo.projectName,
+      projectVersion: documentInfo.projectVersion,
+      entityName: documentReference.metaEdName,
       isDescriptor: false,
     };
 
-    if (foreignKey.isAssignableFrom) {
-      return conditionCheckFromAssignable(entityTypeInfo, foreignKey);
+    if (documentReference.isAssignableFrom) {
```

Comment:
renaming "foreign key" (a relational db concept) to " document reference" in our Meadowlark ubiquitous language